make/tools.mk

```diff
@@ -50,7 +50,7 @@ TOOLS += ko=v0.13.0
 # Version of Gateway API install bundle https://gateway-api.sigs.k8s.io/v1alpha2/guides/#installing-gateway-api
 GATEWAY_API_VERSION=v0.6.2
 
-K8S_CODEGEN_VERSION=v0.27.2
+K8S_CODEGEN_VERSION=v0.29.0
```

Comment:
In https://github.com/cert-manager/cert-manager/pull/7207 @SgtCoDFish also updated to Go 1.21.12, but I propose we  do that in a separate PR with a separate release note:
 * https://github.com/cert-manager/cert-manager/pull/7207/files#diff-99d79958fe5bcd3b582a178172c33191ecfbda3aa8c994e30619960e3b43f58bR59